### DISCUSSION TITLE
Report Malicious Code

### DISCUSSION THREAD

**Discussion by busterbeam** [community/community#36669] [Other Feature Feedback, Questions, & Ideas]
I'm sure there is much more malicious code on Github but I came across [KasRoudra](https://github.com/KasRoudra) / [MaxPhisher](https://github.com/KasRoudra/MaxPhisher).  I've encouraged KasRoudra to update the code so it does not delete the end users home directory.  I've noticed there are forks of this code and I don't think I can get this changed on all the forks, I just would like to get the community to come together to either get all the forks correctly updated of deleted.  I have made [busterbeam](https://github.com/busterbeam)/[MaxPhisher-deobfuscated](https://github.com/busterbeam/MaxPhisher-deobfuscated) which I plan to modify promptly to protect people from accidently having the same fate, but all the code is much easier to read and my commits should have some code to quickly check other repos if they have the same malicious obfuscated code.

Original author of the code says it was to protect against "script kiddies"... but the code simply attacks if you change attribution of code, but I do believe that his GPLv3 license doesn't mean attribution should stay.

ADDITION: PyPhisher is another repo almost identical to MaxPhisher, which has been fixed of malicious code but +100 forks need to be fixed

**Comment by ossanna16** [comment]
Hey there @busterbeam and thank you for reporting this! I'm glad you found our community 🙂

**Reply by busterbeam** [reply]
Anyone going to do anything about this?

**Comment by busterbeam** [comment]
Alright I've called a pull request on all `PyPhisher` and `MaxPhisher` forks to update to a safe version.  Could still do with any more suggestions from the community.

**Comment by jfullstackdev** [comment]
yup, it's better to fix those on both projects, not at GitHub itself, for those malicious code

**Reply by busterbeam** [reply]
Your saying its not github's problem.

**Comment by otter-computer** [comment]
Hi @busterbeam, thanks for the report! In future you can [report these instances directly](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam) to our Trust & Safety team to ensure they see them as soon as possible.

Thanks for keeping the GitHub community safe!

**Reply by busterbeam** [reply]
I did and I still haven't received a response, and I don't see any change amongst these forks.  the bomb is still ticking

**Reply by busterbeam** [reply]
fun fact still no action responded on this almost a year.

**Comment by overtube** [comment]
This repo: `https://github.com/openhardwaremonitor/openhardwaremonitor` when installing on system, using system package manager (that have chroot), after ending installation, post install executing hook that that placing on `/usr/share` directory, **144** `ssh` `root` **certificates**. Offcource code safe and program worked well, but post install activity make malicious, thing because not ask user for trust or not and just wilfully install root certificates.

**Reply by busterbeam** [reply]
It looks like this software gets flagged anyway https://github.com/openhardwaremonitor/openhardwaremonitor/issues/1590.  but really you should open an issue on the repo itself.  I think I've come to accept that you just put it up as an issue on the repo, its the responsibility of users to check recent issues. the software isn't very well documented so you could probably make fork with all the documentation and security fixes. 

please point to exactly where in the code this issue is occurring also.

**Reply by overtube** [reply]
That problem not in code, that trouble happen after software make postinstall hooks without asking any on user, and installs root certificates without any warnings, notifications or asks about thrust or not, that is like a trojan/backdoor installer. Installing software on any system where can be worked **`mono, NET, C#`**

**Reply by busterbeam** [reply]
It must be in the code.  unless this isn't actually the repository used to "build" that application.  I'm no mono expert so I might not be able to find the source of security flaw.  but if you can read c#, find the issue and report it more directly. if you can't make an issue on the actual issue all the same.

**Comment by overtube** [comment]
so, making issue in that case is a wasted. I already create report on github support. So now just wait for answer.

**Reply by busterbeam** [reply]
yeah contacting github directly does nothing if you read what this discussion is really about. if you make an issue on the actual repo other people can read it and be weary of the issue.  but I think this issue may even go beyond that as this application isn't soley on github but they have their own hosted website etc.. if you think it is a serious issue you might need to try and reach out to other "influencers".  plenty of YouTuber's that go over bad software let them no, then they make a video about this software being the devil

**Reply by overtube** [reply]
ok, thx for staging

**Comment by overtube** [comment]
also that seems like a botnet activity prerequisites/preparations or else crap, bc make possible sneak connect on user machine. Fore self purposes, lol so . . .

**Comment by overtube** [comment]
but maybe I was wrong, so
